### PR TITLE
chore(humio_metrics sink): add test for duplicate metrics

### LINKED ISSUE #15619: `humio_metrics` enhanced tag support

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #15619 
Epic #15420 

Humio doesn't really have the concept of metrics as a first class citizen, so duplicate tags should work fine. Since the Humio metrics sink is essentially just a combination of the `metric_to_log` transform and the `humio_logs` sink, this PR just adds a test to ensure that duplicate tags are encoded in a sensible manner (they become an array as per the `metric_to_log` transform).

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

**Comment by spencergilbert** [comment]
Does the sink actually call into the `metric_to_log` transform?

**Comment by bruceg** [comment]
> Does the sink actually call into the `metric_to_log` transform?

FWIW At this point that `metric_to_log` is doing the wrong thing  by outputting array values unconditionally for multi-valued tags. So, any test for such tag values that passes is not using `metric_to_log`.

File: src/sinks/humio/metrics.rs [hunk 1]
```diff
@@ -293,4 +293,47 @@ mod tests {
             output[1].1
         );
     }
+
+    #[tokio::test]
+    async fn duplicate_metrics() {
```
Comment (neuronull) [inline src/sinks/humio/metrics.rs:298]: Isn't this to test duplicate tags?

Also, should we converge on calling it multi-value tags or duplicate tags?

**Comment by StephenWakely** [comment]
> > Does the sink actually call into the `metric_to_log` transform?
> 
> FWIW At this point that `metric_to_log` is doing the wrong thing by outputting array values unconditionally for multi-valued tags. So, any test for such tag values that passes is not using `metric_to_log`.

It is calling into `metric_to_log`. Do you mean that it returns an array for a multi-valued tag that actually only contains a single value?

File: src/sinks/humio/metrics.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/humio/metrics.rs:298]: I've been avoiding the term "duplicate" as the values are not duplicated, only the names.

**Review by bruceg — APPROVED** [review]
I second Kyle's concerns about terms but otherwise this LGTM.
